playlist.c

```diff
@@ -1356,6 +1360,14 @@ bool playlist_push(playlist_t *playlist,
             continue;
       }
 
+#ifdef HAVE_ENTRYSTATES
+      if (playlist->entries[i].state != entry->state)
+      {
+         playlist->entries[i].state   = strdup(entry->state);
```

Comment:
Note that you must `free()` the state path in `playlist_free_entry()`

Reply:
Also gone because of entry state slots beiing unsigned